code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/Optimize.java

```diff
@@ -327,4 +327,77 @@ private static void failWithError(final AdobeCallback<?> callback, final AdobeEr
             callbackWithError.fail(error);
         }
     }
+
+    public static void displayPropositions(List<OptimizeProposition> propositions) {
+        trackWithData(
+                generateInteractionXdmForPropositions(
+                        propositions,
+                        OptimizeConstants.JsonValues.EE_EVENT_TYPE_PROPOSITION_DISPLAY));
+    }
+
+    private static Map<String, Object> generateInteractionXdmForPropositions(
+            List<OptimizeProposition> propositions, final String experienceEventType) {
+        final List<Map<String, Object>> decisioningPropositions =
+                getDecisioningPropositions(propositions);
+        final Map<String, Object> experienceDecisioning = new HashMap<>();
+        experienceDecisioning.put(
+                OptimizeConstants.JsonKeys.DECISIONING_PROPOSITIONS, decisioningPropositions);
+        final Map<String, Object> experience = new HashMap<>();
+        experience.put(OptimizeConstants.JsonKeys.EXPERIENCE_DECISIONING, experienceDecisioning);
+
+        final Map<String, Object> xdm = new HashMap<>();
+        xdm.put(OptimizeConstants.JsonKeys.EXPERIENCE, experience);
+        xdm.put(OptimizeConstants.JsonKeys.EXPERIENCE_EVENT_TYPE, experienceEventType);
+        return xdm;
+    }
+
+    private static @NonNull List<Map<String, Object>> getDecisioningPropositions(
```

Comment:
why do we need a separate method for this? Can we do this inside `generateInteractionXdm`?

Reply:
removed @spoorthipujariadobe